feat(prover-sever): enhance error checking

lightscale-luke opened the pull request:
* update constants related to generate proof. (e.g., MAX_TXS, MAX_CALLDATA, etc)
* check if tx count from trace is smaller than MAX_TXS before starting proof creation.
* check if chain id in trace equals to prover's chain id.
* if requested with an incorrect prove type, prover returns error before initiating a trace object.
* detailed an error message in response.

dongchangYoo committed:
feat(zkevm): apply new values to params

- bump default degree to 21
- increase values for MAX_TXS, MAX_CALLDATA and etc

lightscale-luke committed:
feat(prover-server): validate trace in request before creating proof

- check if chain id in trace equals to prover's chain id.
- check if num of txs in trace is less than MAX_TXS.
- if requested with an incorrect prove type, prover returns error
  before initiating a trace object.
- detailed an error message in response.

fakedev9999 commented on bin/src/prove.rs:
```diff
@@ -78,6 +78,16 @@ fn main() {
+            panic!(
+                "{}",
+                format!(
+                    "Too many transactions. MAX_TXS:{}, given transactions:{}",
```
```suggestion
                    "too many transactions. MAX_TXS:{}, given transactions:{}",
```
let's use lowercase for consistency

fakedev9999 commented on prover-server/src/server_main.rs:
```diff
@@ -58,6 +59,15 @@ impl Rpc for RpcImpl {
+        let tx_count = block_trace.transactions.len();
+        if tx_count > MAX_TXS {
+            kroma_err(format!(
+                "Too many transactions. MAX_TXS:{}, given transactions:{}",
```
```suggestion
                "too many transactions. MAX_TXS:{}, given transactions:{}",
```

fakedev9999 reviewed (approved):
LGTM

dongchangYoo committed:
feat(prover-server): fix bug on receiving response for spec method

dongchangYoo committed:
feat(prover-server): increase request size to 32MiB

adjust the request size to process the JSON trace of a block
containing 100 ERC-20 transfer transactions.

fakedev9999 reviewed (approved):
LGTM

fakedev9999 commented:
@dongchangYoo For the commit message "[feat(prover-server): increase request size to 32Mib](https://github.com/kroma-network/kroma-prover/pull/20/commits/4a934d15108611c2725cea62696003b33302d172)" can you fix it to Mb?

chokobole commented:
Please change commit title `feat: apply new values to params` to `feat(zkevm): apply new values to params`

chokobole commented:
Could you add the reason why you increase request size to 32 Mib in commit body?

dongchangYoo commented:
1 MiB stands for 1,000,000 bytes. However, it would be more accurate to express it as MiB instead of Mib.

dongchangYoo reviewed (approved):
LGTM

fakedev9999 reviewed (approved):
LGTM

chokobole reviewed (approved):
LGTM
